src/Consumer/DoiConsumer.php

```diff
@@ -0,0 +1,310 @@
+<?php
+
+namespace App\Consumer;
+
+use Doctrine\ORM\EntityManagerInterface;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
+
+use PhpAmqpLib\Message\AMQPMessage;
+
+use Symfony\Bridge\Monolog\Logger;
+
+use App\Entity\Dataset;
+use App\Entity\DatasetSubmission;
+use App\Entity\DOI;
+
+use App\EventListener\EntityEventDispatcher;
+
+use App\Util\DOIutil;
+
+use App\Exception\HttpClientErrorException;
+use App\Exception\HttpServerErrorException;
+
+/**
+ * A consumer of DOI messages.
+ *
+ * @see ConsumerInterface
+ */
+class DoiConsumer implements ConsumerInterface
+{
+    /**
+     * The entity manager.
+     *
+     * @var EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * A Monolog logger.
+     *
+     * @var Logger
+     */
+    protected $logger;
+
+    /**
+     * The entity event dispatcher.
+     *
+     * @var EntityEventDispatcher
+     */
+    protected $entityEventDispatcher;
+
+    /**
+     * Delay time in seconds for API.
+     */
+    const DELAY_TIME = 600;
+
+    /**
+     * Constructor.
+     *
+     * @param EntityManagerInterface $entityManager         The entity manager.
+     * @param Logger                 $logger                A Monolog logger.
+     * @param EntityEventDispatcher  $entityEventDispatcher The entity event dispatcher.
+     */
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        Logger $logger,
+        EntityEventDispatcher $entityEventDispatcher
+    ) {
+        $this->entityManager = $entityManager;
+        $this->logger = $logger;
+        $this->entityEventDispatcher = $entityEventDispatcher;
+    }
+
+    /**
+     * Process a filer message.
+     *
+     * @param AMQPMessage $message A filer message.
+     *
+     * @return integer
+     */
+    public function execute(AMQPMessage $message)
+    {
+        // @codingStandardsIgnoreStart
```

Comment:
Is this existing, or was this added? Also this directive will be depricated, it is now phpcs:enable (or something).

Reply:
I think it was existing, should probably remove it

Reply:
It is because of rabbitmq's variable delivery_info, which they haven't yet changed. Will change it to the new implementations of phpcs